sdk/nodejs/query/queryable.ts

```diff
@@ -0,0 +1,54 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { OutputInstance } from "../index";
+import { CustomResource } from "../resource";
+
+export type Queryable<T> = T extends Promise<infer U1>
```

Comment:
It might be nice to have a comment to the effect of what all this typescript dark magic is intended to do.